FirebaseSwiftUI/FirebaseAuthSwiftUI/Sources/FirebaseAuthSwiftUI/FUIAuth.swift

```diff
@@ -0,0 +1,197 @@
+//  var buttonTextColor: UIColor { get }
+//  var buttonAlignment: Alignment { get set }
+  var accessToken: String? { get }
+  var idToken: String? { get }
+
+  // State properties passed upwards
+  var credential: AuthCredential? { get set }
+  var error: Error? { get set }
+  var userInfo: [String: Any]? { get set }
+  var isAuthenticated: Bool { get set }
+
+  func signOut()
+
+  func email() -> String
+
+  // Removed handleOpenURL method as SwiftUI uses onOpenURL which is a view modifier
+}
+
+// similar to FUIAuth in UIKit implementation
+public class FUIAuth: ObservableObject {
+  private var auth: Auth
+  private var authProviders: [FUIAuthProvider] = []
+
+  public init(auth: Auth? = nil) {
+    self.auth = auth ?? Auth.auth()
+  }
+
+  public func authProviders(providers: [FUIAuthProvider]) {
+    authProviders = providers
+  }
+
+  public func getEmailProvider() -> EmailAuthProvider? {
+    return try! providerWithId(providerId: "email") as! EmailAuthProvider
+  }
+
+  public func providerWithId(providerId: String) -> FUIAuthProvider? {
+    if let provider = authProviders.first(where: { $0.providerId == providerId }) {
+      return provider
+    } else {
+      assertionFailure(
+        "Provider with ID \(providerId) not found. Did you add it to the authProviders array?"
+      )
+      return nil
+    }
+  }
+}
+
+// main auth view - can be composed of custom views or fallback to default views. We can also pass
+// state upwards as opposed to having callbacks.
+// Negates the need for a delegate used in UIKit
+public struct FUIAuthView<Modifier: ViewModifier>: View {
+  private var authFUI: FUIAuth
+  private var authPickerView: AuthPickerView<Modifier>
+
+  public init(FUIAuth: FUIAuth,
+              _authPickerView: AuthPickerView<Modifier>? = nil) {
+    authFUI = FUIAuth
+    authPickerView = _authPickerView ?? AuthPickerView()
+  }
+
+  public var body: some View {
+    VStack {
+      AnyView(authPickerView)
+    }
+  }
+}
+
+public protocol AuthPickerViewProtocol: View {
+  var title: AnyView { get }
+}
+
+public struct AuthPickerModifier: ViewModifier {
+  public func body(content: Content) -> some View {
+    content
+      .padding(20)
+      .background(Color.white)
+      .cornerRadius(12)
+      .shadow(radius: 10)
+      .padding()
+  }
+}
+
+public struct AuthPickerView<Modifier: ViewModifier>: AuthPickerViewProtocol {
+  private var emailAuthButton: EmailAuthButton
+  private var vStackModifier: Modifier
+
+  public init(title _: String? = nil, _emailAuthButton: EmailAuthButton? = nil,
+              _modifier: Modifier? = nil) {
+    emailAuthButton = _emailAuthButton ?? EmailAuthButton()
+    vStackModifier = _modifier ?? AuthPickerModifier() as! Modifier
+  }
+
+  public var body: some View {
+    VStack {
+      title
+      emailAuthButton
+    }.modifier(vStackModifier)
+  }
+
+  // Default implementation that can be overridden
+  public var title: AnyView {
+    AnyView(
```

Comment:
Please avoid AnyView. See https://www.swiftbysundell.com/articles/avoiding-anyview-in-swiftui/ and https://youtu.be/YjSxPxT5V40?si=fduquymmiDHmQD86&t=1026.

Annotate the property as `@ViewBuilder` instead. It also probably doesn't need to be `public`, since you're only using it from the `body` of `AuthPickerView`.